modules/apps/feature-flag/feature-flag-test/src/testIntegration/java/com/liferay/feature/flag/test/FeatureFlagListenerTest.java

```diff
@@ -0,0 +1,275 @@
+					!_valueSystem));
+		}
+	}
+
+	@Test
+	public void testRegisterWithMultipleKeys() throws Exception {
+		try (TestFeatureFlagListener testFeatureFlagListener =
+				new TestFeatureFlagListener(
+					_FEATURE_FLAG_KEY_1, _FEATURE_FLAG_KEY_2)) {
+
+			testFeatureFlagListener.assertInvocations(
+				_valuesToString(_companyId, _FEATURE_FLAG_KEY_1, _value1),
+				_valuesToString(_companyId, _FEATURE_FLAG_KEY_2, _value2));
+
+			FeatureFlagTestUtil.setFeatureFlagValue(
+				_companyId, !_value1, _FEATURE_FLAG_KEY_1);
+
+			FeatureFlagTestUtil.setFeatureFlagValue(
+				_companyId, !_value2, _FEATURE_FLAG_KEY_2);
+
+			FeatureFlagTestUtil.setFeatureFlagValue(
+				_companyId, _value1, _FEATURE_FLAG_KEY_1);
+
+			testFeatureFlagListener.assertInvocations(
+				_valuesToString(_companyId, _FEATURE_FLAG_KEY_1, !_value1),
+				_valuesToString(_companyId, _FEATURE_FLAG_KEY_2, !_value2),
+				_valuesToString(_companyId, _FEATURE_FLAG_KEY_1, _value1));
+		}
+	}
+
+	@Test
+	public void testRegisterWithNoKeys() throws Exception {
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				"com.liferay.feature.flag.web.internal.feature.flag." +
+					"FeatureFlagsBagProviderImpl",
+				LoggerTestUtil.ERROR);
+			TestFeatureFlagListener ignored = new TestFeatureFlagListener()) {
+
+			List<LogEntry> logEntries = logCapture.getLogEntries();
+
+			Assert.assertEquals(logEntries.toString(), 1, logEntries.size());
+
+			LogEntry logEntry = logEntries.get(0);
+
+			String message = logEntry.getMessage();
+
+			Assert.assertTrue(
+				message.contains("No featureFlagKey property found for "));
+		}
+	}
+
+	@Test
+	public void testRegisterWithSingleKey() throws Exception {
+		try (TestFeatureFlagListener testFeatureFlagListener =
+				new TestFeatureFlagListener(_FEATURE_FLAG_KEY_1)) {
+
+			testFeatureFlagListener.assertInvocations(
+				_valuesToString(_companyId, _FEATURE_FLAG_KEY_1, _value1));
+
+			FeatureFlagTestUtil.setFeatureFlagValue(
+				_companyId, !_value1, _FEATURE_FLAG_KEY_1);
+
+			FeatureFlagTestUtil.setFeatureFlagValue(
+				_companyId, RandomTestUtil.randomBoolean(),
+				_FEATURE_FLAG_KEY_2);
+
+			testFeatureFlagListener.assertInvocations(
+				_valuesToString(_companyId, _FEATURE_FLAG_KEY_1, !_value1));
+		}
+	}
+
+	@Test
+	public void testRegisterWithWildcardKey() throws Exception {
+		try (TestFeatureFlagListener testFeatureFlagListener =
+				new TestFeatureFlagListener("*")) {
+
+			testFeatureFlagListener.assertSubsetInvocations(
+				_valuesToString(_companyId, _FEATURE_FLAG_KEY_1, _value1),
+				_valuesToString(_companyId, _FEATURE_FLAG_KEY_2, _value2));
+
+			FeatureFlagTestUtil.setFeatureFlagValue(
+				_companyId, !_value1, _FEATURE_FLAG_KEY_1);
+
+			FeatureFlagTestUtil.setFeatureFlagValue(
+				_companyId, !_value2, _FEATURE_FLAG_KEY_2);
+
+			FeatureFlagTestUtil.setFeatureFlagValue(
+				_companyId, _value1, _FEATURE_FLAG_KEY_1);
+
+			testFeatureFlagListener.assertSubsetInvocations(
```

Comment:
Since you are clearing the strings after the subset assertion, this can just be 
`assertInvocations`.